pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -14,15 +14,48 @@
 """
 Class implementation for lightning_gpu state-vector manipulation.
 """
+from warnings import warn
+
+try:
+    from pennylane_lightning.lightning_gpu_ops import StateVectorC64, StateVectorC128
+
+    try:  # Try to import the MPI modules
+        # pylint: disable=no-name-in-module
+        from pennylane_lightning.lightning_gpu_ops import StateVectorMPIC64, StateVectorMPIC128
+
+        MPI_SUPPORT = True
+    except ImportError as ex:
+        warn(str(ex), UserWarning)
+
+        MPI_SUPPORT = False
+
+except ImportError as ex:
+    warn(str(ex), UserWarning)
+
+from itertools import product
 
 import numpy as np
 import pennylane as qml
 from pennylane import DeviceError
+from pennylane.measurements import MidMeasureMP
+from pennylane.ops import Conditional
+from pennylane.ops.op_math import Adjoint
 from pennylane.wires import Wires
 
+from pennylane_lightning.core._serialize import global_phase_diagonal
 from pennylane_lightning.core._state_vector_base import LightningBaseStateVector
 
-from ._measurements import LightningGPUMeasurements
+from ._mpi_handler import MPIHandler
+
+gate_cache_needs_hash = (
```

Comment:
Do we still need this list of gate cache?

Reply:
I am not sure. However, we can come back later on the controlled gates implementation